statement_test.go

```diff
@@ -534,3 +536,60 @@ func TestBindJSON(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, string(jsonData), str)
 }
+
+func TestPrepareComplexQueryParameter(t *testing.T) {
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	ctx := context.Background()
+	createTable(t, db, `CREATE OR REPLACE TABLE arr_test(
+		arr_float float[2]
+	)`)
+
+	// Insert parameters.
+	_, err := db.ExecContext(ctx, `INSERT INTO arr_test VALUES (?)`,
+		[]float32{7, 1})
+	require.NoError(t, err)
+
+	prepared, err := db.Prepare(`SELECT list_value(?,?), list_cosine_distance(?,?)`)
+	defer closePreparedWrapper(t, prepared)
+	require.NoError(t, err)
+
+	var arr Composite[[][]int32]
+	var dis Composite[float32]
+	// Test with specific slice types.
+	err = prepared.QueryRow([]int{1, 2}, []int64{1}, []float32{0.1, 0.2, 0.3}, []float32{0.2, 0.3, 0.4}).Scan(&arr, &dis)
+	require.NoError(t, err)
+	require.Equal(t, [][]int32{{1, 2}, {1}}, arr.Get())
+	require.True(t, dis.Get() > 0)
+
+	dynamicPrepare, err := db.Prepare(`SELECT * from (VALUES (?))`)
+	defer closePreparedWrapper(t, dynamicPrepare)
+	require.NoError(t, err)
+
+	var res Composite[[]any]
+	err = dynamicPrepare.QueryRow([]any{}).Scan(&res)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(res.Get()))
+
+	// The statement type has already been bind as a SQL_NULL LIST in previous query
+	err = dynamicPrepare.QueryRow([]any{1}).Scan(&res)
+	require.NoError(t, err)
+	require.Equal(t, []any{nil}, res.Get())
+
+	ptrPrepare, err := db.Prepare(`SELECT * from (VALUES (?))`)
+	defer closePreparedWrapper(t, ptrPrepare)
+	require.NoError(t, err)
+
+	err = ptrPrepare.QueryRow([]any{toPtr(1)}).Scan(&res)
+	require.NoError(t, err)
+	require.Equal(t, []any{int64(1)}, res.Get())
```

Comment:
Can we add a test with nested pointers? To test the `getPointerValue` a bit more?